packages/plans-grid-next/src/hooks/data-store/types.ts

```diff
@@ -22,6 +22,9 @@ export interface UseGridPlansParams {
 	useCheckPlanAvailabilityForPurchase: Plans.UseCheckPlanAvailabilityForPurchase;
 	useFreeTrialPlanSlugs?: UseFreeTrialPlanSlugs;
 	includePreviousPlanFeatures?: boolean; // if true, each plan would contain the full feature list. if false, the previous plan features are excluded.
+	experimentCallbacks?: {
+		isTrailMapVariant: () => boolean;
+	};
```

Comment:
I don't think that we should introduce Calypso-specific stuff like experiments into the plans grid. Is there way to do this in a parameterized way? One way I can think of is adding `highlightLabelsOverrides` which will be passed to `useGridPlansForFeaturesGrid`. This could be structured like this:
```
highlightLabelsOverrides: ( label, planSlug ) => {
  if ( isTrailMapVariant() && isEcommercePlan( planSlug ) ) {
		return translate( 'Best for eCommerce' );
  }

  return label;
}
```

And then in `useHighlightLabels`:

```
label = highlightLabelsOverrides.?( label, planSlug );
```

Reply:
Yeah, I'm also thinking what @aneeshd16 said above. I think a more modular approach would be for both the feature grid component and the compairson grid component to have a way for the application side to configure which plan to highlight with a badge. That seems to be what we are currently lacking.

There are multiple ways to do this on top of mind at the moment, e.g. 

1. Passing in an `overrides` like @aneeshd16 outlined above.
2. The plans package has default badges for each plan, and the application side can configure which plans to show the badge through a prop.
3. etc. 
 
Whichever way we do, a good general guidance is that it should be constructed in two parts: the functionality of the plans package, and the integration of the plans package and the application, i.e. `plans-features-main` of calypso. The 1st part should be fully testable on Storybook, while the current test plan is for the 2nd part.